subsys/net/lib/openthread/platform/radio.c

```diff
@@ -298,7 +298,13 @@ void transmit_message(struct k_work *tx_job)
 	channel = sTransmitFrame.mChannel;
 
 	radio_api->set_channel(radio_dev, sTransmitFrame.mChannel);
+
+#if defined(CONFIG_IEEE802154_SELECTIVE_TXPOWER)
+	net_pkt_set_ieee802154_txpwr(tx_pkt, get_transmit_power_for_channel(channel));
+	net_pkt_set_ieee802154_txpwr_use(tx_pkt, true);
+#else
 	radio_api->set_txpower(radio_dev, get_transmit_power_for_channel(channel));
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/blob/098602f4aa2460e0b7b61c445b1c5e44a201703b/subsys/net/lib/openthread/platform/radio.c#L648 - I think it should be changed here, too